folsom/src/test/java/com/spotify/folsom/MemcachedServer.java

```diff
@@ -38,34 +40,64 @@ public enum AuthenticationMode {
       Suppliers.memoize(MemcachedServer::new)::get;
 
   public static int DEFAULT_PORT = 11211;
+
+  private static final String MEMCACHED_VERSION = "1.6.21";
   private final String username;
   private final String password;
+  private final boolean useTLS;
 
   public MemcachedServer() {
     this(null, null);
   }
 
   public MemcachedServer(String username, String password) {
-    this(username, password, DEFAULT_PORT, AuthenticationMode.NONE);
+    this(username, password, DEFAULT_PORT, AuthenticationMode.NONE, false);
   }
 
   public MemcachedServer(String username, String password, AuthenticationMode authenticationMode) {
-    this(username, password, DEFAULT_PORT, authenticationMode);
+    this(username, password, DEFAULT_PORT, authenticationMode, false);
+  }
+
+  public MemcachedServer(boolean useTLS) {
+    this(null, null, DEFAULT_PORT, AuthenticationMode.NONE, useTLS);
   }
 
   public MemcachedServer(
-      String username, String password, int port, AuthenticationMode authenticationMode) {
+      String username,
+      String password,
+      int port,
+      AuthenticationMode authenticationMode,
+      boolean useTLS) {
     this.username = username;
     this.password = password;
-    container = new FixedHostPortGenericContainer("bitnami/memcached:1.6.21");
+    this.useTLS = useTLS;
+
+    if (useTLS) {
+      if (authenticationMode != AuthenticationMode.NONE) {
+        throw new RuntimeException(
+            "Authentication is not currently supported with the TLS-enabled container");
```

Comment:
Is this actually true? I can't test it right away, but taking a superficial look at the memcached code, TLS does not disable authentication.

Reply:
That is only about that test class, the issue is with Docker image. "memcached" image supports TLS but not authentication, "bitnami/memcached" supports authentication but not TLS, I was not able to find anything remotely popular that supports both.
I tried to create a Dockerfile based on "memcached" image but enabling SASL authentication doesn't seem trivial and steals the focus from what this PR is aiming to achive, I was going to propose another PR to improve tests after this one is merged.